stytch/core/test/test_models.py

```diff
@@ -108,3 +109,22 @@ def test_stytcherror(self) -> None:
             mock_details = create_autospec(models.StytchErrorDetails)
             str(models.StytchError(mock_details))
             mock_details.__str__.assert_called_once()
+
+    def test_stytcherror_fields(self) -> None:
+        resp = '''{
+          "status_code": 418,
+          "request_id": "request-id-test-fea11c44-5514-4aac-a76b-3ca685e3443a",
+          "error_type": "is_a_teapot",
+          "error_message": "I'm a teapot!",
+          "error_url": "https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418"
+        }'''
+
+        with self.assertRaises(models.StytchError) as e:
+            DummyResponse.from_json(418, json.loads(resp))
+
+        ex = e.exception
+        self.assertEqual(ex.details.error_type, "is_a_teapot")
+        self.assertEqual(ex.details.status_code, 418)
+        self.assertEqual(ex.details.error_url, "https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418")
+        self.assertEqual(ex.details.error_message, "I'm a teapot!")
+        self.assertEqual(ex.details.request_id, "request-id-test-fea11c44-5514-4aac-a76b-3ca685e3443a")
```

Comment:
[pebble] Store your expected details in an `expected = StytchErrorDetails(...)` object above and then just call `self.assertEqual(expected, e.exception.details)` here.

A strategy I've used and advocated for in the past is the [Arrange-Act-Assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) test pattern. You'll see in some instances in this codebase where I've made that explicit with comments like `# Arrange`, but I also don't feel strongly about having the implicit separation. Either way, I do think it would be nice to create an `expected` object and just do a single `assertEqual` since `pydantic` will implement `__eq__` for us nicely.

Reply:
d719d8544907e96c40a07b1326145759d4114669

I feel like I may have gone a step too far and obscured the purpose of the test with `**resp`. It's easy to unroll that next time we find ourselves here. It still catches the bug it's supposed to!

Originally, I was trying to emphasize how a real consumer would check one of these fields, but the unittest raise helper makes the code too different from real app code. I thought about adding another test in real-app style (explicit try-except with `==`) but I think that's probably better to add in the Readme or other docs.